tests/helpers/util.lua

```diff
@@ -0,0 +1,75 @@
+local utils = {}
+
+local Path = require("plenary.path")
+local testdir = Path:new(vim.env.TMPDIR or "/tmp", "neo-tree-testing"):absolute()
+
+local function rm_test_dir()
+  if vim.fn.isdirectory(testdir) == 1 then
+    vim.fn.delete(testdir, "rf")
+  end
+end
+
+utils.setup_test_fs = function()
+  rm_test_dir()
+
+  -- Need a list-style map here to ensure that things happen in the correct order.
+  --
+  -- When/if editing this, be cautious as (for now) other tests might be accessing
+  -- files from within this array by index
+  local fs = {
```

Comment:
Should there be a wrapper function for this, such that you can request a file from the tree in a simpler way (that is also more resistant to change in this list)?

Right now you have to:

```lua
local fs = utils.setup_test_fs()
local testfile = fs.content[3].abspath
```

which is obviously not the best thing because as soon as someone messes with the ordering here, all of those references break. One idea might be to give each file a "handle", and access it by the handle, so that a wrapper function can access it like:

```lua
local fs = utils.setup_test_ts()
local testfile = fs.get("top_level_file")
```

which would return a file that suffices the criteria that is being asked for. You could have handles like `"empty_dir"`, `"nested_file"` and whatnot which might make this a bit better? This turns into a "yo dawg" issue, where you need tests for your tests and so on.